tests/common/collapse.rs

```diff
@@ -104,3 +104,24 @@ where
     collapser.collapse(r, std::io::sink()).unwrap();
     test_logger::validate(asserter);
 }
+
+pub fn test_collapse_error<C>(mut collapser: C, test_filename: &str) -> io::Error
```

Comment:
Are there other collapsers that could use this right now?

Reply:
Same here, I'll use it for the sample collapser.